oligo_designer_toolsuite/database/_oligo_database.py

```diff
@@ -721,58 +575,140 @@ def get_sequence_oligoid_mapping(
-    ############################################
+    def get_oligo_attribute_value(
+        self, attribute: str, flatten: bool, region_id: str, oligo_id: str
+    ) -> Union[str, List[str]]:
+
+        if not region_id in self.database:
+            raise ValueError(f"Region {region_id} does not exist.")
+
+        if not oligo_id in self.database[region_id]:
+            raise ValueError(f"Region {oligo_id} does not exist.")
+
+        oligo_attributes = self.database[region_id][oligo_id]
+        if attribute not in oligo_attributes:
+            attribute_value = None
+        elif flatten:
+            attribute_value = flatten_attribute_list(self.database[region_id][oligo_id][attribute])
+        else:
+            attribute_value = self.database[region_id][oligo_id][attribute]
 
-    def update_oligo_attributes(self, new_oligo_attribute: dict):
-        """Update attributes of oligonucleotides in the database with new values.
+        return attribute_value
 
-        Iterates through each oligonucleotide in the database and updates its attributes based on the provided new attribute values.
-        The update is done in place, modifying the existing attributes of the oligonucleotides.
+    ############################################
+    # Manipulation Functions
+    ############################################
 
-        Input format of new attribute:
-        new_attribute = {"region_x::1": {"attribute_y": 110}}
+    def update_oligo_attributes(self, new_oligo_attribute: dict) -> None:
 
-        :param new_oligo_attribute: A dictionary mapping oligonucleotide IDs to their new attributes.
-        :type new_oligo_attribute: dict
-        """
         for region_id, database_region in self.database.items():
             for oligo_id, oligo_attributes in database_region.items():
-                oligo_attributes.update(new_oligo_attribute[oligo_id])
+                if oligo_id in new_oligo_attribute:
+                    oligo_attributes.update(format_oligo_attributes(new_oligo_attribute[oligo_id]))
+
+    def filter_database_by_region(self, remove_region: bool, region_ids: Union[str, List[str]]) -> None:
+
+        # Check formatting
+        region_ids = check_if_list(region_ids)
+        if self.database:
+            for region_id in self.database.keys():
+                if remove_region and (region_id in region_ids):
+                    del self.database[region_id]
+                elif not remove_region and (region_id not in region_ids):
+                    del self.database[region_id]
+        else:
+            raise ValueError(
+                "Can not filter. Database is empty! Call the method load_database() or load_database_from_fasta() first."
+            )
+
+    def filter_database_by_oligo(self, remove_region: bool, oligo_ids: Union[str, List[str]]) -> None:
+
+        # Check formatting
+        oligo_ids = check_if_list(oligo_ids)
+        if self.database:
+            for region_id in self.database.keys():
```

Comment:
The code could be parallelized region-wise.

Reply:
True, we will address it in the oligo attribute issue